### PR TITLE
Day11

### PR DESCRIPTION
**Pull request by Pilchie**
I did *not* come up with any elegant solutions  here.  I *also* didn't get the debugger working, which was a bit of a PITA.

### CODE REVIEW

**Comment by Pilchie** [comment]
@anurse

File: 2020/11/seating/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+            occupied += self.one_if_occupied(y - 1, x + i - 1);
+            occupied += self.one_if_occupied(y + 1, x + i - 1);
+        }
+        occupied += self.one_if_occupied(y, x - 1);
+        occupied += self.one_if_occupied(y, x + 1);
+
+        occupied
+    }
+
+    fn visible_occupied(&self, y: usize, x: usize) -> usize {
```
Comment (anurse) [inline 2020/11/seating/src/main.rs:104]: Raycasting! Now you can be a GPU programmer ;P

You should be able to have a helper that takes in a "ray" (`dx`, `dy` pair) and finds the point of the visible occupied seat. I did something like [this](https://github.com/anurse/advent-of-code/blob/main/2020/day11/day11.go#L170) to reduce the repetition.

File: 2020/11/seating/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+            },
+            _ => cur,
+        }
+    }
+
+    fn adjacent_occupied(&self, y: isize, x: isize) -> usize {
+        let mut occupied = 0;
+
+        for i in 0..3 {
+            occupied += self.one_if_occupied(y - 1, x + i - 1);
```
Comment (anurse) [inline 2020/11/seating/src/main.rs:95]: If you wanted to get *super* zesty, you could implement `Into<usize>` for `Position` and then `self.one_if_occupied(x, y)` just becomes `self.get(x, y).into()`

File: 2020/11/seating/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+    pub fn occupied(&self) -> usize {
+        let mut count = 0;
+        for x in &self.map {
+            for y in x {
+                count += match y {
+                    Position::Occupied => 1,
+                    _ => 0,
+                }
+            }
+        }
```
Comment (anurse) [inline 2020/11/seating/src/main.rs:249]: Definitely some combinators on `Iterator` that could make this functiony :). The lack of generics and thus any useful functional-programming operators is driving me nuts doing this in Go :(.

**Pull request merged**